src/app/chat/chat.component.ts

```diff
@@ -1,8 +1,117 @@
+  private chatColors:any[];
+  private colorArray = [];
+
+  constructor(private chatService: ChatService, private navCtrl: NavController, private storage: Storage) {
+  }
+
+  ngOnInit() {
+    this.haveMoreTeam = false;
+    this.loadChatData();
+  }
+
+  loadChatData():void {
+    this.chatService.getchatList().subscribe(response => {
+      this.updateChatListResponse(response);
+    });
+  }
+
+  /**
+   * modify the response 
+   *  - set chat avatar color
+   * @param {Array} response 
+   */
+  private updateChatListResponse(response):void {
+    if ((response)) {
+      this.chatList = [];
+      // let chatColors = this.chatService.getChatAvatarColors();
+      if ((!this.chatColors)) {
+        this.setChatAvatarColors(response, null, 'nocolor');
+      } else {
+        this.setChatAvatarColors(response, this.chatColors, 'havecolor');
+      }
+      this.checkHaveMoreTeam(response);
+    }
+  }
+
+  /**
+   * this method check old avatar colors and set the releted one.
+   * if there no old avatar colors it will create new color and add that to service variable.
+   * @param {Array} response 
+   * @param {Array} chatColors 
+   * @param {String} status 
+   */
+  private setChatAvatarColors(response, chatColors, status):void {
+    let index = 0;
+    for (index = 0; index < response.length; index++) {
+      if ((response[index])) {
+        switch (status) {
+          case "nocolor":
+            response[index].chat_color = this.chatService.getRandomColor();
+            this.colorArray.push({
+              team_member_id: response[index].team_member_id,
+              name: response[index].name,
+              chat_color: response[index].chat_color
+            });
+            break;
+          case "havecolor":
+            let colorObject = chatColors.find(function(chat) {
+              return chat.team_member_id === response[index].team_member_id;
+            });
+            if ((colorObject)) {
+              response[index].chat_color = colorObject.chat_color;
+            } else {
+              response[index].chat_color = this.chatService.getRandomColor();
+            }
+            break;
+        }
+        this.chatList.push(response[index]);
+      }
+    }
+    if ((this.colorArray.length > 0)) {
+      this.storage.set('chatAvatarColors', this.colorArray);
+    }
+  }
+
+  /**
+   * this method check is this user in multiple teams.
+   * @param {Array} response 
+   */
+  private checkHaveMoreTeam(response):void {
+    let index = 0;
+    let teamCount = 0;
+    for (index = 0; index < response.length; index++) {
+      if ((response[index].is_team)) {
+        teamCount++;
+      }
+    }
+    if ((teamCount > 1)) {
+      this.haveMoreTeam = true;
+    } else {
+      this.haveMoreTeam = false;
+    }
+  }
+
+  getChatAvatarText(chatName) {
+    return this.chatService.generateChatAvatarText(chatName);
+  }
+
+  navigateToChatRoom(chat) {
+    this.storage.set('selectedChatObject', chat);
```

Comment:
It seems you just saved the chat object to local storage, and use this as a reference for the chat-room page.
The correct way of doing this is passing data in the route instead of save it to local storage. 
The route of the chat-room page should contain enough information to load the correct chat message. 
Only by doing that we can implement deep linking to every page (including the chat-room page)
Let's say we make the route of the chat-room like this `/chat/chat-room/:teamId/:teamMemberId`, because `team id` is necessary for every chat message. The logic is: if there's a `teamMemberId` passed in, it is a person to person message. If there's no `teamMemberId` passed in, it is a team message.